stratum/lib/security/credentials_manager_test.cc

```diff
@@ -0,0 +1,172 @@
+// Copyright 2021-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+#include "stratum/lib/security/credentials_manager.h"
+
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "absl/memory/memory.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/substitute.h"
+#include "gmock/gmock.h"
+#include "grpcpp/grpcpp.h"
+#include "gtest/gtest.h"
+#include "stratum/glue/net_util/ports.h"
+#include "stratum/glue/status/status_test_util.h"
+#include "stratum/lib/security/test.grpc.pb.h"
+#include "stratum/lib/utils.h"
+
+DECLARE_string(ca_cert);
+DECLARE_string(server_key);
+DECLARE_string(server_cert);
+DECLARE_string(test_tmpdir);
+
+namespace stratum {
+namespace {
+
+class TestServiceImpl final : public ::testing::TestService::Service {
+ public:
+  ::grpc::Status Test(::grpc::ServerContext* /*context*/,
+                      const ::testing::Empty* /*request*/,
+                      ::testing::Empty* /*response*/) override {
+    return ::grpc::Status::OK;
+  }
+};
+
+enum class CertFile {
+  kCaCert = 0,
+  kServerCert = 1,
+  kServerKey = 2,
+};
+
+constexpr char kCertDirPrefix[] = "stratum/lib/security/testdata/certs";
+constexpr char kCaCertFile[] = "ca.crt";
+constexpr char kServerCertFile[] = "stratum.crt";
+constexpr char kServerKeyFile[] = "stratum.key";
+
+std::string GetFilename(CertFile f, int i) {
+  const char* file;
+  switch (f) {
+    case CertFile::kCaCert:
+      file = kCaCertFile;
+      break;
+    case CertFile::kServerCert:
+      file = kServerCertFile;
+      break;
+    case CertFile::kServerKey:
+      file = kServerKeyFile;
+      break;
+  }
+  return absl::StrFormat("%s%d/%s", kCertDirPrefix, i, file);
+}
```

Comment:
just hardcode the four paths?

Reply:
This method is gone as of the latest commit - let me know what you think of the latest code, but I'll resolve this comment for now